dropshot/tests/fail/bad_endpoint10.stderr

```diff
@@ -1,17 +1,22 @@
-error[E0271]: type mismatch resolving `<String as TypeEq>::This == HttpError`
+error[E0277]: the trait bound `Result<HttpResponseOk<()>, String>: HttpResponse` is not satisfied
   --> tests/fail/bad_endpoint10.rs:16:6
    |
 16 | ) -> Result<HttpResponseOk<()>, String> {
-   |      ^^^^^^
-   |      |
-   |      expected this to be `HttpError`
-   |      type mismatch resolving `<String as TypeEq>::This == HttpError`
+   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `HttpResponse` is not implemented for `Result<HttpResponseOk<()>, String>`
```

Comment:
This seems like a step backward; could `#[diagnostic]` help here?

Reply:
What do you think of the new error message in https://github.com/oxidecomputer/dropshot/pull/1180/commits/f9642d16f850869f98fdec2cc51528f0ba4f07cc?